csrc/ir/internal_nodes.h

```diff
@@ -129,6 +129,59 @@ class IndexSelectOp : public Expr {
   }
 };
 
+class IndexPutAccumulateOp : public Expr {
+ public:
+  using Expr::Expr;
+
+  // logical ID groups of IndexPutAccumulateOp
+  // args:
+  //     acc   [ ID_indexed_g0, ID_g0 ]
```

Comment:
Indexed IDs in the existing indirect access ops mean those IDs that are indirectly accessed IDs. Here, `ID_indexed_g0` is the indexed ID, but `ID_indexed_g1` doesn't seem so, does it?

Reply:
Ah, I think this naming confusion comes from the use of `getIndexedDomainInfo`. As the comment of the function says, it is meant to be used when producer tensors are accessed through index tensors, which isn't the case with index_put.

Reply:
Yeah naming is tricky with this one.

I think `ID_indexed_g0` should remain as-is. Meanwhile, `ID_indexed_g1` should be renamed as `ID_indexing_g1` instead, since that's the ID used to indexing into `ID_indexed_g0`.

So the other function below `getIndexedID`->`getIndexingID`.

As for `getIndexedDomainInfo`, we don't have to name the function that way. It's really trying to return IDs that are indirectly accessed on the producer. So I'll rename it as `getIndirectlyAccessedIDs`.

For the other index ops, those are the indexed IDs, but for index_put, it's the indexing IDs.

Reply:
> but for index_put, it's the indexing IDs.

IIUC, the indexing IDs of the index and value tensors are not indirectly accessed, so `getIndirectyAccessedIDs` doesn't seem to a right name.

Reply:
For my own curiosity, so `indirectly accessed` means the ID doesn't get a contiguous access in the generated code via indexing. Then I agree it's not the right name for that.

As for the function `getIndexedDomainInfo`, I'll renaming it as `getNonMappingDomainInfo`. In the case of indexPutAccumulate, the indexing IDs of the index and value doesn't map to output.
It's less informative a name, I'll note the cases as comment in the code.